### PR TITLE
Fix: Node CLI arguments for dafny to have the right order

### LINKED ISSUE #2867: Behaviour of Dafny program command-line arguments with "dafny build" inconsistent on Javascript
```dafny
// An attempt at a CLI that behaves the same in all target languages
method Main(args: seq<string>) {
  if |args| == 0 {
    return;
  }
  print args[1..], "\n";
}
```

```
> dafny /compile:3 src/Scratch.dfy /compileTarget:js --args 1 2 3

Dafny program verifier finished with 0 verified, 0 errors
Running...

1,2,3
> dafny build -t:js src/Scratch.dfy

Dafny program verifier finished with 1 verified, 0 errors
> a483e79acc1e:libraries salkeldr$ node src/Scratch.js 1 2 3
/Users/salkeldr/Documents/GitHub/libraries/src/Scratch.js,1,2,3
```

### PR DESCRIPTION
**Pull request by MikaelMayer**
This PR fixes #2867
Tests for C#, JavaScript and C++
I tried to spend some time to make it work for Java but it seems we don't have a standard procedure for launching programs, so I leave it unspecified for now.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

**Review by robin-aws — DISMISSED** [review]
Nice, thanks for the quick fix! My only ask is to explicitly call this out as a `fix:` release note. This could be considered a breaking change, but I think it's reasonable to say it was underspecified and this is clearly better behaviour.

> I tried to spend some time to make it work for Java but it seems we don't have a standard procedure for launching programs, so I leave it unspecified for now.

That's fine for this PR, but FYI https://github.com/dafny-lang/dafny/blob/master/Test/comp/manualcompile/ManualCompile.dfy has the blueprint you're looking for. But I would also love to add support for command-line arguments to `%testDafnyForEachCompiler` :)

**Comment by MikaelMayer** [comment]
> That's fine for this PR, but FYI https://github.com/dafny-lang/dafny/blob/master/Test/comp/manualcompile/ManualCompile.dfy has the blueprint you're looking for. But I would also love to add support for command-line arguments to `%testDafnyForEachCompiler` :)

Oh thanks, I did not know about this file ! That was very helpful.
I was able to add the tests for the remaining 3 back-ends in no time !

File: docs/dev/news/2876.fix [hunk 1]
```diff
@@ -0,0 +1 @@
+Same argument order for stand-alone javascript files than for other languages
```
Comment (keyboardDrummer) [inline docs/dev/news/2876.fix:1]: than => as

**Review by keyboardDrummer — DISMISSED** [review]

File: docs/dev/news/2876.fix (hunk 1 above)
Comment (MikaelMayer) [inline docs/dev/news/2876.fix:1]:
```suggestion
Same argument order for stand-alone javascript files as for other languages
```

**Review by keyboardDrummer — APPROVED** [review]

**Comment by cpitclaudel** [comment]
Can we refine the release notes for this?  It's hard to understand what this means without context (and reading the PR description doesn't help too much either)

**Pull request merged**